packages/squiggle-lang/src/PointSet/Mixed.ts

```diff
@@ -232,7 +243,11 @@ export class MixedShape implements PointSet<MixedShape> {
     const discreteMean = this.discrete.mean();
     const continuousMean = this.continuous.mean();
     // means are already weighted by subshape probabilities
-    return (discreteMean + continuousMean) / this.integralSum();
+    return (
+      (discreteMean * this.discrete.integralSum() +
+        continuousMean * this.continuous.integralSum()) /
+      this.integralSum()
+    );
```

Comment:
This is an important fix, old code was incorrect.

Reply:
Makes sense